spec/models/theme_field_spec.rb

```diff
@@ -775,4 +775,57 @@ def create_yaml_field(value)
       expect(theme.scss_variables).not_to include("theme_uploads")
     end
   end
+
+  describe "migration JavaScript field" do
+    it "must match a specific format for filename" do
+      field = Fabricate(:migration_theme_field, theme: theme)
+      field.name = "12-some-name"
+
+      expect(field.save).to eq(false)
```

Comment:
```suggestion
      expect(field.valid?).to eq(false)
```

Just a small tip but you can use `valid?` instead of trying to save the field.

Reply:
Agreed! I've switched to `valid?` here and in the other spots.